homeassistant/components/shelly/__init__.py

```diff
@@ -140,6 +137,46 @@ def __init__(self, hass, entry, device: aioshelly.Device):
 
         self.device.subscribe_updates(self.async_set_updated_data)
 
+        self._async_remove_input_events_handler = self.async_add_listener(
+            self._async_input_events_handler
+        )
+        self._last_input_events_count = dict()
+
+    @callback
+    def _async_input_events_handler(self):
+        """Handle device input events."""
+        for block in self.device.blocks:
+            if "inputEvent" and "inputEventCnt" in block.sensor_ids:
```

Comment:
Make sure you rewrite code with guard clauses. 
```suggestion
            if "inputEvent" not in block.sensor_ids or "inputEventCnt" not in block.sensor_ids:
                continue
```